src/pe/optional_header.rs

```diff
@@ -249,6 +284,24 @@ impl OptionalHeader {
             magic => Err(error::Error::BadMagic(u64::from(magic))),
         }
     }
+
+    /// Validate the Optional Header.
+    pub fn validate(&self) -> error::Result<()> {
+        let mut error_messages = vec![];
+        if let Err(error::Error::Malformed(error_message)) = self.windows_fields.validate() {
```

Comment:
While you're here you might as well validate IMAGE_OPTIONAL_HEADER.Magic == IMAGE_NT_OPTIONAL_HDR_MAGIC

https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_optional_header32